src/states/proving_node.rs

```diff
@@ -0,0 +1,408 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::common::{Base, Bootstrapped};
+use crate::{
+    ack_manager::AckManager,
+    action::Action,
+    cache::Cache,
+    config_handler,
+    error::RoutingError,
+    event::Event,
+    id::{FullId, PublicId},
+    messages::{DirectMessage, MessageContent, RoutingMessage},
+    outbox::{EventBox, EventBuf},
+    peer_manager::{Peer, PeerManager, PeerState},
+    resource_prover::ResourceProver,
+    routing_message_filter::RoutingMessageFilter,
+    routing_table::{Authority, Prefix},
+    state_machine::Transition,
+    timer::Timer,
+    types::{MessageId, RoutingActionSender},
+    xor_name::XorName,
+    CrustEvent, PubConnectionInfo, Service,
+};
+use maidsafe_utilities::serialisation;
+use std::{
+    collections::BTreeSet,
+    fmt::{self, Display, Formatter},
+};
+
+pub struct ProvingNode {
```

Comment:
Adam, could you point me how you created this file (proving_node.rs), where did you pick the pieces from (since it is not removed in this commit)

Reply:
The pieces are mostly extracted from `node.rs`, but they are removed from it only in later commits. It was easier for me to work that way but I can see how it can make the reviewing harder. Sorry about that :flushed:

Reply:
No problem, just needed the right source to compare.